### PR TITLE
Feature/single file first

### PR DESCRIPTION
**Pull request by niklas**
When a watched test file is changed, run it with mix test. If the unix
error code is 0 (successful), run the complete suite.
When the file is not in the test directory, try to find a corresponding
test file following elixir conventions and run that instead.

This should speed up TDD extremely.

--

* replaces #21 (accidental merge)
* will be squashed when considered supreme

### CODE REVIEW

**Comment by niklas** [comment]
build this please

**Comment by lpil** [comment]
I think this behaviour should be opt in, rather than on by default.

**Comment by praveenperera** [comment]
Hey what's the status on this pull request? This would be an awesome feature to have and I've tested this particular branch in some of my projects and it seems to working fine.

**Comment by lpil** [comment]
My comment above needs to be taken into account before it can be merged :)

**Comment by praveenperera** [comment]
@niklas - any plans on making this opt-in? If not can I work on this?

**Comment by lpil** [comment]
I should point out that this will be very difficult to merge after #29 and #30 are merged. Might be better to write it from scratch.

File: lib/mix_test_watch/logger.ex [hunk 1]
```diff
@@ -0,0 +1,11 @@
+defmodule MixTestWatch.Logger do
```
Comment (lpil) [inline lib/mix_test_watch/logger.ex]: Is this module used anywhere?

**Comment by niklas** [comment]
I am going to have a look at this PR in this week or the next. I am not sure on how to enable/disable the feature. Command line arguments? Configuration? What's the "Elixir way"?

**Comment by praveenperera** [comment]
I would say setting it in the config would be the most idiomatic elixir.

**Comment by lpil** [comment]
I agree.

File: lib/mix_test_watch/run.ex [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  @spec run_tests(String.t) :: :ok
+
+  def run_tests(config) do
+    config |> Command.build |> Shell.exec
+  end
+
+  @spec run_one(String.t, String.t) :: :ok
+
```
Comment (daveshah) [inline lib/mix_test_watch/run.ex]: Are these the correct typing for the spec? Maybe I'm mis-reading (or a little naive when it comes to typespecs) but it looks like we're returning a tuple below with `{:one, ... }` but in the typespec above defines the return as `:ok`. Is that correct?

File: lib/mix_test_watch/files.ex [hunk 3]
```diff
@@ -0,0 +1,19 @@
+defmodule MixTestWatch.Files do
+  @moduledoc """
+  Finds corresponding test file for any given file.
+  """
+
+  @spec find_test(String.t) :: String.t
```
Comment (daveshah) [inline lib/mix_test_watch/files.ex]: Should this be `@spec find_test(String.t) :: {:ok, String.t} ` ?

**Comment by lpil** [comment]
I feel with the `--stale` flag there is no need for this any more.

**Pull request closed without merging**